docs/shared_bindings_matrix.py

```diff
@@ -166,21 +169,40 @@ def get_excluded_boards(base):
             if board_chip in port_config:
                 contents += "\n" + "\n".join(port_config[board_chip])
 
+            check_dependent_modules = dict()
             for module in modules:
                 board_is_excluded = False
                 # check if board uses `SMALL_BUILD`. if yes, and current
                 # module is marked as `FULL_BUILD`, board is excluded
                 small_build = re.search("CIRCUITPY_SMALL_BUILD = 1", contents)
                 if small_build and base[module]["full_build"] == "1":
                     board_is_excluded = True
+
+                # check if board uses `MINIMAL_BUILD`. if yes, and current
+                # module is marked as `DEFAULT_BUILD`, board is excluded
+                min_build = re.search("CIRCUITPY_MINIMAL_BUILD = 1", contents)
+                if min_build and base[module]["default_value"] == "CIRCUITPY_DEFAULT_BUILD":
+                    board_is_excluded = True
+
                 # check if module is specifically disabled for this board
                 re_pattern = "CIRCUITPY_{}\s=\s(\w)".format(module.upper())
                 find_module = re.search(re_pattern, contents)
                 if not find_module:
-                    # check if default inclusion is off ('0'). if the board doesn't
-                    # have it explicitly enabled, its excluded.
-                    if base[module]["default_value"] == "0":
-                        board_is_excluded = True
+                    if base[module]["default_value"].isdigit():
+                        # check if default inclusion is off ('0'). if the board doesn't
+                        # have it explicitly enabled, its excluded.
+                        if base[module]["default_value"] == "0":
+                            board_is_excluded = True
+                    else:
+                        # this module is dependent on another module. add it
+                        # to the list to check after processing all other modules.
+                        # only need to check exclusion if it isn't already excluded.
+                        if (not board_is_excluded and
+                            base[module]["default_value"] not in [
+                                "None",
+                                "CIRCUITPY_DEFAULT_BUILD"
```

Comment:
What are the possible values of "default_value" at this point? Looking at build_module_map it will be `1` or `"None"` since 0 is false and int() is used to convert the value to an int.

Reply:
The only real outliers at this point in time are `audiocore` and `audiomixer` (https://github.com/adafruit/circuitpython/blob/master/py/circuitpy_mpconfig.mk#L88).

- `audiocore`: `default_value = "CIRCUITPY_AUDIOPWMIO, CIRCUITPY_AUDIOIO"`.
- `audiomixer`: `default_value = "CIRCUITPY_AUDIOIO"`.

I imagine `audiosynth` will follow the same suit.

#2021 lists a couple other outliers that have not been addressed yet, at the `py/circuitpy_mpconfig.mk` level.

Reply:
Ah, ok. I was misreading the code above.